libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -62,9 +62,15 @@ void NavEKF3_core::controlMagYawReset()
     bool finalResetRequest = false;
     bool interimResetRequest = false;
     if (flightResetAllowed && !assume_zero_sideslip()) {
+#if APM_BUILD_TYPE(APM_BUILD_ArduSub)
+        // for sub, we'd like to be far enough away from metal structures like docks and vessels
+        // diving below 2.5m is a reasonable test, so we can use the same constant, but the sign is flipped
+        finalResetRequest = (stateStruct.position.z  - posDownAtTakeoff) > EKF3_MAG_FINAL_RESET_ALT;
```

Comment:
could we use 0.5m for sub instead? that makes it easier to test on shallower tanks/pools (mine is 1.5m deep)